Migration to Next 12

davidsingal opened the pull request:
Migrating to Next 12, instead of 13 because it requires React 18+ and it will require more changes in the origin code.

* Upgrade react and react-dom to version 17.0.2, minimun version required for Next 12.

clementprdhomme reviewed (changes requested):
In addition to my comment below, I've found a few issues that weren't present on production.

## Navigating to a journey

When the user navigates to a journey, the page stays blank. If the user reloads, then the journey is shown.

[After navigating to a journey, the page stays blank](https://user-images.githubusercontent.com/6073968/225638859-93c13e52-5b8d-48be-b12f-d1be21c70c9a.webm)

Similarly, when the user reaches the end of a journey and clicks the link to open the following, the page stays blank.

[At the end of the journey, after clicking the link to go the the next one, the page stays blank](https://user-images.githubusercontent.com/6073968/225639735-b6d6c786-b584-441a-991e-f202bfbe2c41.webm)

## Map position lost

When the user clicks the link to open a journey map as a standalone tab, the position of the map is lost.

[When opening a journey embedded map as a standalone tab, the map's position is lost](https://user-images.githubusercontent.com/6073968/225640586-094d6c89-1c57-4ca4-8d84-5c5b2b380476.webm)

## Login feature

It seems the login feature is not working correctly. After the user logs in, they are not redirected to the homepage. If they reload the page, they have to log in again. Even if they don't reload the page, it seems they are not correctly logged in because no personal information is visible.

[After logging in, the user is not redirected. If they go see their profile, it's empty. If they reload the page, they're not logged in anymore.](https://user-images.githubusercontent.com/6073968/225641746-8e72e885-0deb-4c2e-9d78-d6bb7e86c251.webm)

davidsingal commented on frontend/docker-compose.yml:
```diff
@@ -5,7 +5,7 @@ services:
       dockerfile: Dockerfile
       args:
-        - REACT_APP_API_HOST=https://www.resilienceatlas.org
+        - NEXT_PUBLIC_API_HOST=https://www.resilienceatlas.org
```
we could but it's not needed to work, it will be just undefined, but I'll add it anyway

clementprdhomme commented on frontend/tsconfig.tsbuildinfo:
```diff
@@ -0,0 +1 @@
+{"program":{"fileNames":["./node_modules/typescript/lib/lib.es5.d.ts","./node_modules/typescript/lib/lib.es2015.d.ts","./node_modules/typescript/lib/lib.es2016.d.ts","./node_modules/typescript/lib/lib.es2017.d.ts","./node_modules/typescript/lib/lib.es2018.d.ts","./node_modules/typescript/lib/lib.es2019.d.ts","./node_modules/typescript/lib/lib.es2020.d.ts","./node_modules/typescript/lib/lib.es2021 ...
```
I think this file can be added to `.gitignore`: https://github.com/vercel/next.js/discussions/30815

davidsingal replied:
you are right, on it

clementprdhomme reviewed (commented):
In addition to my comment below, I noticed a difference between the previous code and this new one.

Before, after the user would sign in, they would get redirected to the homepage. In this PR, this is not the case anymore. Could you have a look?

davidsingal commented:
> In addition to my comment below, I noticed a difference between the previous code and this new one.
> 
> Before, after the user would sign in, they would get redirected to the homepage. In this PR, this is not the case anymore. Could you have a look?

Done, added the redirect. Also, I've fixed the profile fetch. But it seems like it was not working before.

clementprdhomme reviewed (approved):
Feel free to merge. Really good job!
